Running `make verify` on after `make install_static` throws tons of linker errors

## Description
The `make verify` step, when run after installing GoCV with statically built OpenCV, throws tons of linker errors, such as:

```
go run ./cmd/version/main.go
# gocv.io/x/gocv
/usr/bin/ld: /usr/local/lib/libopencv_objdetect.a(qrcode.cpp.o): in function `cv::QRDecode::decodingProcess() [clone .part.0]':
qrcode.cpp:(.text._ZN2cv8QRDecode15decodingProcessEv.part.0+0x10d): undefined reference to `quirc_decode'
/usr/bin/ld: /usr/local/lib/libopencv_imgcodecs.a(grfmt_exr.cpp.o): in function `cv::ExrDecoder::readHeader()':
grfmt_exr.cpp:(.text._ZN2cv10ExrDecoder10readHeaderEv+0x12): undefined reference to `Imf_opencv::globalThreadCount()'
/usr/bin/ld: grfmt_exr.cpp:(.text._ZN2cv10ExrDecoder10readHeaderEv+0x30): undefined reference to `Imf_opencv::InputFile::InputFile(char const*, int)'


< ... some 9400 lines cut out due to size constraints ... >


/usr/bin/ld: /usr/local/lib/libopencv_core.a(filesystem.cpp.o): in function `cv::plugin::impl::DynamicLib::libraryRelease()':
filesystem.cpp:(.text._ZN2cv6plugin4impl10DynamicLib14libraryReleaseEv+0x434): undefined reference to `dlclose'
/usr/bin/ld: /usr/local/lib/libopencv_dnn.a(types.pb.cc.o): in function `opencv_tensorflow::DataType_descriptor()':
types.pb.cc:(.text._ZN17opencv_tensorflow19DataType_descriptorEv+0xe): undefined reference to `google::protobuf::internal::AssignDescriptors(google::protobuf::internal::DescriptorTable const*, bool)'
/usr/bin/ld: /usr/local/lib/libopencv_dnn.a(types.pb.cc.o): in function `_GLOBAL__sub_I.00102_types.pb.cc':
types.pb.cc:(.text.startup._GLOBAL__sub_I.00102_types.pb.cc+0xf): undefined reference to `google::protobuf::internal::AddDescriptorsRunner::AddDescriptorsRunner(google::protobuf::internal::DescriptorTable const*)'
collect2: error: ld returned 1 exit status
make: *** [Makefile:277: verify] Error 2
```

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Check out the repository
2. Run `make install_static`

## Your Environment
<!--- Include as many relevant details about your environment -->

* Operating System and version: Debian 11 64 bit
* OpenCV version used: 4.5.5 (as installed by GoCV 0.30.0 Makefile)
* How did you install OpenCV: Via Makefile
* GoCV version used: 0.30.0
* Go version: go1.18 linux/amd64
* Did you run the `env.sh` or `env.cmd` script before trying to `go run` or `go build`? Yes and no. Tried both with and without with the same result.

--- a/Makefile
+++ b/Makefile
@@ -250,7 +250,7 @@ install_cuda: deps download sudo_pre_install_clean build_cuda sudo_install clean
 install_openvino: deps download download_openvino sudo_pre_install_clean build_openvino_package sudo_install_openvino build_openvino sudo_install clean verify_openvino
 
 # Do everything statically.
-install_static: deps download sudo_pre_install_clean build_static sudo_install clean verify
+install_static: deps download sudo_pre_install_clean build_static sudo_install clean verify_static
 
 # Do everything with non-free modules from cpencv_contrib.
 install_nonfree: deps download sudo_pre_install_clean build_nonfree sudo_install clean verify
@@ -275,6 +275,10 @@ sudo_install_openvino:
 # Build a minimal Go app to confirm gocv works.
 verify:
 	go run ./cmd/version/main.go
+
+# Build a minimal Go app to confirm gocv works with statically built OpenCV.
+verify_static:
+	go run -tags static ./cmd/version/main.go
 
 # Build a minimal Go app to confirm gocv cuda works.
 verify_cuda:

